Update GC documents (#3006)
pingcap/docs: garbage-collection-configuration.md

- Specifies the GC concurrency manually. This parameter works only when you set [`tikv_gc_auto_concurrency`](#tikv_gc_auto_concurrency) to `false`.
- Default: 2

## `tikv_gc_scan_lock_mode` (**experimental feature**)

> **Note:**
>
> Green GC is still an experimental feature. It is recommended **NOT** to use it in the production environment.

> This parameter specifies the way of scanning locks in the Resolve Locks step of GC, which means whether or not to enable Green GC (experimental feature). In the Resolve Locks step of GC, TiKV needs to scan all locks in the cluster. With Green GC disabled, TiDB scan locks in the unit of Regions. Green GC provides the "physical scanning" feature, which means that each TiKV node can bypass the Raft layer to directly scan data. This feature can effectively mitigate the impact of GC wakening up all Regions when the [Hibernate Region](/tikv-configuration-file.md#raftstorehibernate-regions-experimental) feature is enabled, thus improving the execution speed in the Resolve Locks step.

Reviewer: ```suggestion
This parameter specifies the way of scanning locks in the Resolve Locks step of GC, that is, whether to enable Green GC (experimental feature) or not. In the Resolve Locks step of GC, TiKV needs to scan all locks in the cluster. With Green GC disabled, TiDB scans locks by Regions. Green GC provides the "physical scanning" feature, which means that each TiKV node can bypass the Raft layer to directly scan data. This feature can effectively mitigate the impact of GC wakening up all Regions when the [Hibernate Region](/tikv-configuration-file.md#raftstorehibernate-regions-experimental) feature is enabled, thus improving the execution speed in the Resolve Locks step.
```

> - `"legacy"` (default): Uses the old way of scanning, which means to disable Green GC.

Reviewer: ```suggestion
- `"legacy"` (default): Uses the old way of scanning, that is, disable Green GC.
```

> - `"physical"`: Uses the physical scanning method, which means to enable Green GC.

Reviewer: ```suggestion
- `"physical"`: Uses the physical scanning method, that is, enable Green GC.
```

- Auto-concurrent: requests are sent to each Region concurrently with the number of TiKV nodes as concurrency value.
- Distributed: no need for TiDB to send requests to TiKV to trigger GC because each TiKV handles GC on its own.

> In addition, if Green GC (experimental feature) is enabled, which means setting the value of [`tikv_gc_scan_lock_mode`](#tikv_gc_scan_lock_mode-experimental-feature) to `physical`, the processing of GC is not affected by the configuration above.

Reviewer: ```suggestion
In addition, if Green GC (experimental feature) is enabled, that is, setting the value of [`tikv_gc_scan_lock_mode`](#tikv_gc_scan_lock_mode-experimental-feature) to `physical`, the processing of Resolve Lock is not affected by the concurrency configuration above.
```
